### PR TITLE
rcache: Allow disabling of rcache

### PR DESCRIPTION
**Pull request by andrewwormald**
This allows the ability to disable the cache at runtime. Primarily can be used for testing but can also have applications outside of testing. This change would force the behaviour of RCache to allow all calls to read through to the underlying loader.

### CODE REVIEW

**Comment by adamhicks** [comment]
Is there any way to leverage the existing option on EventsTable `WithoutEventsCache` to achieve your goals?

**Comment by andrewwormald** [comment]
The issue with using that is that there is real option to have it different at runtime. Happy to make it testing exclusive?

File: rsql/rcache.go [hunk 1]
```diff
@@ -22,6 +22,18 @@ type rcache struct {
 	limit  int
 }
 
+// enabled is the source of truth whether the rcache is enabled or not.
+var enabled bool
```
Comment (NeilLuno) [inline rsql/rcache.go:26]: This variable is called rsql.enabled. We're not disabling "rsql" - we're disabling just the cache part. Suggest `var disableCache bool`.

File: rsql/rcache.go [hunk 2]
```diff
@@ -22,6 +22,18 @@ type rcache struct {
...
+
+func init() {
+	enabled = true
```
Comment (NeilLuno) [inline rsql/rcache.go:29]: Don't need this.

File: rsql/rcache.go [hunk 3]
```diff
@@ -62,7 +74,8 @@ func (c *rcache) tailUnsafe() int64 {
 func (c *rcache) Load(ctx context.Context, dbc *sql.DB,
 	prev int64, lag time.Duration,
 ) ([]*reflex.Event, error) {
-	if res, ok := c.maybeHit(prev+1, lag); ok {
+	res, ok := c.maybeHit(prev+1, lag)
+	if enabled && ok {
```
Comment (NeilLuno) [inline rsql/rcache.go:78]: If the cache isn't enabled we don't even need to call maybeHit. Suggest 

```go
if !disableCache {
  return c.readThrough(ctx, dbc, prev, lag)
}
```

File: rsql/rcache.go (hunk 1 above)
Comment (andrewwormald) [reply rsql/rcache.go:26]: Ah good point.

File: rsql/rcache.go (hunk 3 above)
Comment (andrewwormald) [reply rsql/rcache.go:78]: I was trying to avoid using negatives as its generally seen as less readable

Comment (NeilLuno) [reply rsql/rcache.go:78]: I think the zero value (false) should be the default. If you want to override that thing, it should be true. BTW I got the logic wrong - that should be `if disableCache {}`.

**Comment by NeilLuno** [comment]
> Is there any way to leverage the existing option on EventsTable `WithoutEventsCache` to achieve your goals?

We looked into that together. The way we initialise events tables makes it tricky. It seems more maintainable to me to turn the cache off globally in a test, rather than turning the cache off for specific events tables in each test case.

File: rsql/rcache.go (hunk 3 above)
Comment (andrewwormald) [reply rsql/rcache.go]: Havn't gone gone quite with this recommendation as I wanted the checks to be the same both outside of `readThrough` and the check inside `readThrough`

**Review by NeilLuno — APPROVED** [review]

**Review by NeilLuno — APPROVED** [review]

**Pull request merged**